bitcoin/src/crypto/sighash.rs

```diff
@@ -1245,6 +1242,39 @@ impl<'a> Encodable for Annex<'a> {
     }
 }
 
+/// Annex must be at least one byte long and the first bytes must be `0x50`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum AnnexError {
+    /// The annex is empty.
+    Empty,
+    /// Incorrect prefix byte in the annex.
+    IncorrectPrefix(u8),
+}
+
+impl fmt::Display for AnnexError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use AnnexError::*;
+
+        match *self {
+            Empty => write!(f, "the annex is empty"),
+            IncorrectPrefix(byte) =>
+                write!(f, "incorrect prefix byte in the annex {:02x}, expecting 0x50", byte),
```

Comment:
Damn I didn't notice it's still inconsistent because of 0x, but a minor issue, not going to hold up the PR.

Reply:
Yeah, oops. But agreed, no need to hold up the PR. At least we're not mixing bases.